### DISCUSSION TITLE
无效的404路由只显示Main，不显示菜单栏和标题栏

### DISCUSSION THREAD

**Discussion by WeiJunFenYou** [dotnetcore/BootstrapBlazor#3992] [Q&A] [answered]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Describe the bug

![image](https://github.com/user-attachments/assets/2c984fb3-31f0-4a50-9412-275d2809c3a6)

点击进入这个地址，就会只显示Main，而不渲染菜单栏和标题栏。直接从浏览器输入无效的URL也是如此。

![image](https://github.com/user-attachments/assets/439bc57a-d3e3-41ce-a4c9-f05c4109cd76)

在2年前的老版本中，就没有这个问题。即便URL无效，也会正确的渲染它们。

页面的代码如下。
`            
<NotFound>
                <LayoutView Layout="@typeof(DefaultLayout)">
                    <p>Sorry, there's nothing at this address.</p>
                </LayoutView>
            </NotFound>
`
我怀疑这是个BUG，但我没有证据。有可能是因为我对组件掌握不够，不过这个问题困扰了我许久，还望开发者解惑。

### .NET Version

.NET 8

**Comment by bb-auto** [comment]
@WeiJunFenYou Thank you for filing this issue. In order for us to investigate this issue, please provide [a minimalistic repro project](https://github.com/dotnetcore/bootstrapblazor/blob/main/docs/repro.md) that illustrates the problem.

**Comment by ArgoZhang** [comment]
@WeiJunFenYou 请提供简单的复现例子，而不是 abp 例子，请帮忙点一下右上角 star 谢谢啦

**Comment by WeiJunFenYou** [comment]
好的。我使用官网提供的项目模板，创建了一个“BootstrapBlazor.UiBugTest”项目。这是一个 Server side类型的项目。源代码在这里：https://github.com/WeiJunFenYou/BootstrapBlazor.UiBugTest

![image](https://github.com/user-attachments/assets/b223e3be-7aca-4893-9fd5-eb05dfe30cd9)
![image](https://github.com/user-attachments/assets/eecb1119-9e53-4e1a-94bd-9412a6eadc01)

如果访问一个无效的url地址，代码中预设的行为，是显示“Sorry, there's nothing at this address.”。但模板项目却显示无法访问。

预期的结果，应该是程序正确的渲染标题栏和菜单栏，并显示“Sorry, there's nothing at this address.”
@ArgoZhang

**Comment by ArgoZhang** [comment]
@WeiJunFenYou 感谢，我复现如果是 bug 我修复一下

**Comment by ArgoZhang** [comment, marked as answer]
@WeiJunFenYou `net8` 移除了 `NotFound` 模板，更改为使用 `Error` 页面

https://learn.microsoft.com/en-us/aspnet/core/fundamentals/error-handling?view=aspnetcore-8.0

所以目前就是这样的

```
if (app.Environment.IsDevelopment())
{
    app.UseDeveloperExceptionPage();
}
else
{
    app.UseExceptionHandler("/Error");
}
```


`Error` 页面里需要自己处理相关逻辑

```
public async Task OnGet()
{
    RequestId = Activity.Current?.Id ?? HttpContext.TraceIdentifier;

    var exception = HttpContext.Features.Get<IExceptionHandlerFeature>();
    if (exception?.Endpoint?.DisplayName == "/_Host" && exception?.Error is CryptographicException)
    {
        var adminUrl = await _dict.GetAdminUrl();
        if (!string.IsNullOrEmpty(adminUrl))
        {
            adminUrl = string.Format(adminUrl, Request.Scheme, Request.Host.Host);
            Response.Redirect(adminUrl);
        }
    }
}
```

**Reply by WeiJunFenYou** [reply]
既然移除了，为什么在项目中仍然可以使用NotFound标签而不产生编译错误？

**Reply by WeiJunFenYou** [reply]
意思是说，接下来的项目模板，以及使用此库的用户们，以后都不再使用NotFound标签来处理404错误了，而是使用Error 页面这种新方式。是这个意思吗。代码中已有的NotFound标签要删掉，改成Error 页面？

**Reply by ArgoZhang** [reply]
@WeiJunFenYou 看一下微软文档吧，都有写，我们模板这样弄是从 `net6` 迁移上来的，兼容老项目

**Reply by ArgoZhang** [reply]
@WeiJunFenYou 帮忙选个回答最为答案可以吗？这样方便后来人直接找到答案，感谢感谢
